### DISCUSSION TITLE
IDAT too much image data

### DISCUSSION THREAD

**Discussion by vaidyasaurabh08** [ImageMagick/ImageMagick#7106] [Help]
### ImageMagick version

7.1.1-20

### Operating system, version and so on

linux

### Description

Getting following error on converting an image:

convert: cache resources exhausted 
```
convert: IDAT: Too much image data `image.png' @ warning/png.c/MagickPNGWarningHandler/1526.
convert: cache resources exhausted `image.png' @ error/cache.c/OpenPixelCache/3906.
convert: cache resources exhausted `image.png' @ error/cache.c/OpenPixelCache/3906.
convert: No IDATs written into file `image.png' @ error/png.c/MagickPNGErrorHandler/1492.
```

The image details are as follows using the identify command :
format=PNG height=9000 width=5400

Image size is a mere 1897672 bytes that is 1.8 MB
I see the resource limits should be sufficient to handle this image:

```
Resource limits:
  Width: 13000P
  Height: 13000P
  Area: 512MP
  List length: unlimited
  Memory: 512MiB
  Map: 512MiB
  Disk: 1GiB
  File: 768
  Thread: 1
  Throttle: 1
  Time: 300
```

Trying to identify which limit is it that is blocking this image from "convert"

**Comment by fmw42** [comment]
Post the exact command you used to convert the image. Post the input image.

**Comment by snibgo** [comment]
> Image size is a mere 1897672 bytes that is 1.8 MB

That is the file size, in bytes, of the PNG compression. That isn't relevant to the problem.

I suppose you have run out of memory. How much physical memory do you have free when running IM?

Your image has about 45 million pixels. What version of IM are you running? Q-number? HDRI? You might need about 500 MB for the pixel cache, which is about the limit of both memory and map. But I suspect you have run out of memory within the PNG converter.

You might try "-limit memory 1MB -limit map 1MB" before reading the input file. This will ensure the cache doesn't eat all your memory.
